optaplanner-examples/src/main/java/org/optaplanner/examples/examination/optional/score/ExaminationConstraintProvider.java

```diff
@@ -0,0 +1,371 @@
+                .penalize("Two exams in a row", HardSoftScore.ONE_SOFT, (topicConflict, leftExam, rightExam) -> topicConflict.getStudentSize());
+    }
+
+    //    // Two exams in a day which share students
+//    rule "twoExamsInADay"
+//    when
+//    $topicConflict : TopicConflict($leftTopic : leftTopic, $rightTopic : rightTopic)
+//    $leftExam : Exam(topic == $leftTopic, $leftDayIndex : dayIndex, $leftPeriodIndex : periodIndex, period != null)
+//    $rightExam : Exam(topic == $rightTopic, dayIndex == $leftDayIndex,
+//                      Math.abs($leftPeriodIndex - periodIndex) > 1)
+//    then
+//        scoreHolder.penalize(kcontext, $topicConflict.getStudentSize());
+//    end
+    private Constraint twoExamsInADay(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(TopicConflict.class)
+                .join(Exam.class,
+                      equal(TopicConflict::getLeftTopic, Exam::getTopic))
+                .filter((topicConflict, leftExam) -> leftExam.getPeriod() != null)
+                .join(Exam.class,
+                      equal((topicConflict, leftExam) -> topicConflict.getRightTopic(), Exam::getTopic),
+                      equal((topicConflict, leftExam) -> leftExam.getDayIndex(), Exam::getDayIndex))
+                .filter((topicConflict, leftExam, rightExam) -> Math.abs(leftExam.getPeriodIndex() - rightExam.getPeriodIndex()) > 1)
+                .penalize("Two exams in a day",
+                          HardSoftScore.ONE_SOFT,
+                          (topicConflict, leftExam, rightExam) -> topicConflict.getStudentSize());
+    }
+
+    //    // Exams which share students have to few periods between them
+//    rule "periodSpread"
+//    when
+//    $constraintConfiguration : ExaminationConstraintConfiguration()
+//    $topicConflict : TopicConflict($leftTopic : leftTopic, $rightTopic : rightTopic)
+//    $leftExam : Exam(topic == $leftTopic, $leftPeriodIndex : periodIndex, period != null)
+//    $rightExam : Exam(topic == $rightTopic, period != null,
+//                      Math.abs($leftPeriodIndex - periodIndex) < ($constraintConfiguration.getPeriodSpreadLength() + 1))
+//    then
+//        scoreHolder.penalize(kcontext, $topicConflict.getStudentSize());
+//    end
+    private Constraint periodSpread(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(TopicConflict.class)
+                .join(Exam.class,
+                      equal(TopicConflict::getLeftTopic, Exam::getTopic))
+                .filter((topicConflict, leftExam) -> leftExam.getPeriod() != null)
+                .join(Exam.class,
+                      equal((topicConflict, leftExam) -> topicConflict.getRightTopic(), Exam::getTopic),
+                      equal((topicConflict, leftExam) -> leftExam.getDayIndex(), Exam::getDayIndex))
+                .filter((topicConflict, leftExam, rightExam) -> Math.abs(leftExam.getPeriodIndex() - rightExam.getPeriodIndex()) > 1)
```

Comment:
You use this `Math.abs(...) >/= 1` construct frequently. I'd extract it to a separate method and reuse.

Reply:
Also same comment regarding `ifExists()` as with `twoExamsInADay`.